substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -609,9 +569,13 @@ pub mod pallet {
 	/// Rewards for the last [`Config::HistoryDepth`] eras.
 	/// If reward hasn't been set or has been removed then 0 reward is returned.
 	#[pallet::storage]
-	#[pallet::unbounded]
-	pub type ErasRewardPoints<T: Config> =
-		StorageMap<_, Twox64Concat, EraIndex, EraRewardPoints<T::AccountId>, ValueQuery>;
+	pub type ErasRewardPoints<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		EraRewardPoints<T::AccountId, <T::ElectionProvider as ElectionProviderBase>::MaxWinners>,
```

Comment:
Actually this is not correct. The storage `ValidatorCount` is the right value. 

We can potentially turn it into a config item but that has the implication that increasing validator count would always require runtime upgrade and cannot happen via governance. To fix that, you will have to also turn it into a dynamic parameter (see [here](https://github.com/polkadot-fellows/runtimes/blob/main/relay/kusama/src/lib.rs#L637)).

Reply:
I removed `MaxValidatorsCount` from storage and added a `MaxValidatorsCount` dynamic parameter. I think it's better to have only one of them.